plugins/rest/unit_test/test_utils.py

```diff
@@ -111,6 +114,26 @@ def test_get_json_decode_error(self, mock_get):
 
         self.assertIn("I am a teapot", e.exception.data.msg)
 
+    @mock.patch("requests.request", side_effect=mocked_requests_get)
+    def test_get_data_string_is_unencoded(self, mock_get):
+        data = "client_id=12345&client_secret=passwd"
+        expected = "client_id=12345&client_secret=passwd"
+        log = logging.getLogger("Test")
+        api = RestAPI("www.httpbin.org", log, True, {})
+        result = api.call_api("get", "/", data)
+        result = result.data
+        self.assertEqual(expected, result)
+
+    @mock.patch("requests.request", side_effect=mocked_requests_get)
+    def test_get_data_string_is_encoded(self, mock_get):
+        data = "client_id=12345&client_secret=passwd"
+        expected = "client_id=12345&client_secret=passwd"
+        log = logging.getLogger("Test")
+        api = RestAPI("www.httpbin.org", log, True, {})
+        result = api.call_api("get", "/", data, headers={"Content-Type": "application/x-www-form-urlencoded"})
+        result = result.data
+        self.assertEqual(expected, result)
+
```

Comment:
https://pypi.org/project/parameterized/
let's use parametrized if tests are similar

Reply:
Here are a couple of examples of using parameterized in a unit test suite:
https://github.com/rapid7/insightconnect-plugins/blob/master/plugins/fortinet_fortigate/unit_test/test_get_policies.py
https://github.com/rapid7/insightconnect-plugins/blob/master/plugins/microsoft_log_analytics/unit_test/test_search.py